devops/azure-pipelines.yml

```diff
@@ -0,0 +1,84 @@
+# This pipeline relies on the analysis.yml and dockerize.yml templates 
+# located in the repository resource descibed below
+# The ref: value must point to a branch where both templates exist.
+resources:
+  repositories:
+  - repository: templates
+    type: github
+    endpoint: Team1-Project3
+    name: 1053-August-Duet-Project-Registry/project-registry-devops
+    ref: refs/heads/feature-dockerpipe
+
+# This pipeline will trigger on commits or pull-requests made to the branch-dev 
+# or main branches.  Note that complete static analysis with SonarCloud is only 
+# performed on "long-lived" branches, designated as 'main' and any branch name 
+# prefixed with 'branch-'.  A minimal analysis is still performed on "short-lived" 
+# branches, and a special analysis is performed on pull requests.
+trigger:
+- branch-dev
+- main
+pr:
+- branch-dev
+- main
+
+pool:
+  vmImage: ubuntu-latest
+
+# These variable groups are required for sonarCloud and Discord secrets to be
+# available in the analysis template
+# The contents of these groups that are used here are:
+#  project-registry-sonarcloud: stores sonarcloud variables/secrets
+#   - sonarKeyPrefix: the common prefix string for the sonar keys in the project
+#       (requires a sonarCloud project key)
+#   - sonarNamePrefix: the common prefix string for the project names
+#   - sonarOrg: the sonar organization
+#   - sonarUrlPrefix: the common prefix string for the url for the results
+#  project-registry-discord: stores discord variables/secrets
+#   - discordAnalysisChannel: the output channel for discord analysis info
+#   - discordAnalysisKey: the webhook key associated with the above channel
+variables:
+- group: project-registry-sonarcloud
+- group: project-registry-discord
+
+stages:
+- stage: analyze
+  jobs:
+  - job: analyze
+
+    steps:
+    - checkout: self
+    - checkout: templates
+    - template: ./templates/analysis.yml@templates
+      parameters:
+        microservice: gateway
+        workDir: ./project-registry-gateway/
+        sonarKeyPrefix: $(sonarKeyPrefix)
+        sonarNamePrefix: $(sonarNamePrefix)
+        sonarOrg: $(sonarOrg)
+        sonarUrlPrefix: $(sonarUrlPrefix)
+        discordChannel: $(discordAnalysisChannel)
+        discordKey: $(discordAnalysisKey)
+
+    displayName: Analyze --> Job
+
+  displayName: Analyze --> Stage
+
+- stage: dockerize
+  dependsOn: [] # <-- run concurrent with analyze stage; does not depend on it
+  jobs:
+  
+  - job: dockerize
+    steps:
+    - template: ./templates/dockerize.yml@templates
+      parameters:
+        imageName: project-registry-gateway
+        containerRepo: jpbulloch5
+        containerServiceConnection: 'Jacob''s Docker Hub'
+        containerUrl: https://hub.docker.com/repository/docker/jpbulloch5/project-registry-gateway
+        discordChannel: $(discordDockerChannel)
+        discordKey: $(discordDockerKey)
```

Comment:
we should be able to change this to use the acr now, but if not it can wait

Reply:
Rich wants us to stick to the old one until we are ready to work with pushing to AKS.